src/ppl/src/IOPlacer.cpp

```diff
@@ -755,6 +755,8 @@ bool IOPlacer::assignPinsToSections(int assigned_pins_count)
 
   total_pins_assigned += assigned_pins_count;
 
+  assert(net->numIOPins() >= total_pins_assigned);
```

Comment:
Not sure an `assert` is the right thing. If it's an error then we should generate an error with a human readable string, or a warning if a warning is appropriate.

Reply:
This is simply a bug in the application, so I think an assert() is appropriate.

I think about errors as appropriate when the  application is correctly implemented, but cant create a result for some reason.

Reply:
Asserts can be compiled out but errors can't.  Also after an assert you can't inspect the application state.

Reply:
On a large design it may be too slow to run in a debugger conveniently.  If it happens in a release build it is still bad so why wouldn't you want the check?

Reply:
All good. I will change it to an error.